src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/CreateCohortController.cs

```diff
@@ -35,20 +53,41 @@ public IActionResult Index(IndexRequest request)
         [Route("select-provider")]
         public IActionResult SelectProvider(SelectProviderRequest request)
         {
-            var viewModel = new SelectProviderViewModel();//todo: from mapper
+            var viewModel = _selectProviderViewModelMapper.Map(request);
 
             return View(viewModel);
         }
 
         [Route("select-provider")]
         [HttpPost]
-        public IActionResult SelectProvider(SelectProviderViewModel request)
+        public async Task<IActionResult> SelectProvider(SelectProviderViewModel request)
         {
-            //todo:hit api
+            GetProviderResponse providerResponse;
+            try
+            {
+                var validationResult = _selectProviderViewModelValidator.Validate(request);
 
-            var confirmProviderRequest = new ConfirmProviderRequest();//todo: from mapper
+                if (!validationResult.IsValid)
+                {
+                    return View(request);
+                }
 
-            return RedirectToAction("ConfirmProvider", confirmProviderRequest);
+                providerResponse = await _commitmentsApiClient.GetProvider(long.Parse(request.ProviderId));
+                
+                var confirmProviderRequest = _confirmProviderRequestMapper.Map(request);
+
+                return RedirectToAction("ConfirmProvider", confirmProviderRequest);
+            }
+            catch (RestHttpClientException e)
+            {
+                if (e.StatusCode == HttpStatusCode.NotFound)
+                {
+                    ModelState.AddModelError(nameof(providerResponse.ProviderId), "Check UK Provider Reference Number");
+                    return View(request);
+                }
```

Comment:
else ... ?? other RestHttpClientExceptions won't get logged or anything, which would not be good. also worth having another catch for exception and redirect to the error page.

Reply:
Added catch for unexpected exceptions which log and return the error view